File: src/my_cpp_pkg/src/topics/smart_phone.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class SmartPhoneNode : public rclcpp::Node // MODIFY NAME
{
public:
    SmartPhoneNode() : Node("smart_phone") // MODIFY NAME
    {
        subscriber_ = this->create_subscription<std_msgs::msg::String>(
            "robot_news",
            10,
            std::bind(&SmartPhoneNode::callbackRobotNews, this, std::placeholders::_1)
        );
        RCLCPP_INFO(this->get_logger(), "Smart phone has been started (new).");
    }

private:
    std::shared_ptr<rclcpp::Subscription<std_msgs::msg::String>> subscriber_;

    void callbackRobotNews(const std_msgs::msg::String msg_){
        RCLCPP_INFO(this->get_logger(), "Subscribed to news: '%s'", msg_.data.c_str());
        
    }
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmartPhoneNode>(); // MODIFY NAME
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/topics/smart_phone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Smart_Phone_Node(Node): # MODIFY NAME
    def __init__(self):
        super().__init__("smart_phone") # MODIFY NAME
        self.subscriber_ = self.create_subscription(String, "robot_news", self.smart_phone_callback, 10)
       

    def smart_phone_callback(self, msg: String):
        self.get_logger().info(f"Received message: {msg.data}")


def main(args=None):
    rclpy.init(args=args)
    node = Smart_Phone_Node() # MODIFY NAME
    node.get_logger().info("smart_phone node is activated!")
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
